### PR TITLE
Replace `block` mode with `sync` when tx broadcasting

### LINKED ISSUE #164: Use `sync` or `async` instead of `block` to broadcast transaction
https://github.com/informalsystems/atomkraft/blob/84e4a377fc749f2e50f21ce25f46f8ff812d3733/atomkraft/chain/testnet.py#L396-L401

Tx broadcasting in `block` mode is deprecated. The block inclusion should be inspected via work done in informalsystems/atomkraft#86.

### PR DESCRIPTION
**Pull request by rnbguy**
Closes #164

### CODE REVIEW

File: atomkraft/chain/testnet.py [hunk 1]
```diff
@@ -384,20 +384,38 @@ def broadcast_transaction(
...
+                # # BROADCAST_MODE_ASYNC defines a tx broadcasting mode where the client
+                # # returns immediately.
+                # BROADCAST_MODE_ASYNC = 3
+                result = asyncio.run(
+                    service.broadcast_tx(
+                        tx_bytes=bytes(tx.to_proto()),
+                        mode=BroadcastMode.BROADCAST_MODE_SYNC,
+                    )
+                ).tx_response
+                if result.code == 0:
```
Comment (hvanz) [inline atomkraft/chain/testnet.py:397]: The line `with TmEventSubscribe({"tm.event": "Tx"}) as subscriber:` should be here, just before `subscriber` is used.

File: atomkraft/chain/testnet.py [hunk 2]
```diff
@@ -384,20 +384,38 @@ def broadcast_transaction(
...
-            # # returns immediately.
-            # BROADCAST_MODE_ASYNC = 3
-            result = asyncio.run(
-                service.broadcast_tx(
-                    tx_bytes=bytes(tx.to_proto()),
-                    mode=BroadcastMode.BROADCAST_MODE_BLOCK,
-                )
-            ).tx_response
+            with TmEventSubscribe({"tm.event": "Tx"}) as subscriber:
+                # # BROADCAST_MODE_BLOCK defines a tx broadcasting mode where the client waits
```
Comment (hvanz) [inline atomkraft/chain/testnet.py]: I would remove most of the comments, leaving only the relevant one about sync mode.

**Review by hvanz — APPROVED** [review]
Tested manually and works as expected.

File: atomkraft/chain/testnet.py (hunk 1 above)
Comment (rnbguy) [reply atomkraft/chain/testnet.py:397]: Good comment :smile: but this is needed. `TmEventSubscribe` can not subscribe to past events. If we broadcast a transaction and we are too late to subscribe, the intended transaction will not be recorded.

File: atomkraft/chain/testnet.py [hunk 3]
```diff
@@ -384,20 +384,38 @@ def broadcast_transaction(
...
+                # # for the tx to be committed in a block.
+                # BROADCAST_MODE_BLOCK = 1
+                # # BROADCAST_MODE_SYNC defines a tx broadcasting mode where the client waits
+                # # for a CheckTx execution response only.
+                # BROADCAST_MODE_SYNC = 2
+                # # BROADCAST_MODE_ASYNC defines a tx broadcasting mode where the client
+                # # returns immediately.
+                # BROADCAST_MODE_ASYNC = 3
```
Comment (rnbguy) [inline atomkraft/chain/testnet.py]:
```suggestion
                # # BROADCAST_MODE_SYNC defines a tx broadcasting mode where the client waits
                # # for a CheckTx execution response only.
                # BROADCAST_MODE_SYNC = 2
```

Comment (rnbguy) [reply atomkraft/chain/testnet.py]: @hvanz does this look alright?

Comment (hvanz) [reply atomkraft/chain/testnet.py]: Yes, thanks

**Pull request merged**